### ISSUE TITLE
CA1508 false positive with ??-pattern

### ISSUE THREAD

**Issue by SebastianSchumann** [dotnet/roslyn-analyzers#6695]
### Analyzer

**Diagnostic ID**: [CA1508](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1508): `Avoid dead conditional code`

### Analyzer source

**SDK**: Built-in CA analyzers in .NET 7 SDK

**Version**: SDK 7.0.304

### Describe the bug

False positive when using ?? pattern:

```c#
private void FailingMethod(IEnumerable<int>? items = null)
{
	var itemsToTest = items as IReadOnlyList<int> ??
					  items?.ToList() ??
					  (IReadOnlyList<int>)Array.Empty<int>();
}
```

### Steps To Reproduce

Add this method somewhere to your code and try to compile using all warnings as error.

### Expected behavior

No warning.

### Actual behavior

The line `items?.ToList()` produces `Error CA1508: 'items' is always 'null'. Remove or refactor the condition(s) to avoid dead code.`
